CHANGELOG.md

```diff
@@ -1,317 +1,853 @@
-  * Expose `.options` property on LevelUP instances. (@rvagg)
-  * Further clarify 'encoding' -> 'valueEncoding' shift. db.options.valueEncoding is now authoritative even if user used 'encoding' on initialisation. (@rvagg)
-  * 'level' package now published to npm that bundles LevelUP & LevelDOWN and exposes LevelUP directly; for planned shift to detaching LevelDOWN as a direct-dependency of LevelUP. (@rvagg)
+### Added
+* README: add node version badge (@ralphtheninja)
+* Add Typescript definitions and testing (@MeirionHughes)
 
-## 0.7.0 @ 8 Apr 2013
+### Changed
+* README: homogenize readme style (@vweevers)
+* Update `level-errors` (@ralphtheninja)
+* Optimize Typescript tests (@vweevers)
 
-  * Windows support in LevelDOWN @0.2.0 (@rvagg)
-  * added 'db' option on constructor to replace LevelDOWN (@rvagg)
-  * added repair() & destroy() aliases for LevelDOWN implementations (@rvagg)
-  * fix early 'close' emit in WriteStream (@rvagg)
-  * improved ReadStream reverse=true start key handling (@kesla)
-  * ReadStream empty start & end keys ignored rather than segfault (@kesla)
-  * 'encoding' option now an alias for 'valueEncoding' only, 'keyEncoding' defaults to 'utf8' and must be changed explicitly (@rvagg)
+### Removed
+* Remove 7 from Travis (@ralphtheninja)
 
-## 0.6.2 @ 4 Mar 2013
+## [2.0.0-rc1] - 2017-09-01
 
-  * use `xtend` package instead of internal util._extend @ralphtheninja
-  * internal cleanup of `callback` argument detection @ralphtheninja
-  * move deferred-open-operations into an internal `this._db` wrapper rather than make them call public .get()/.put() etc. for a second time @dominictarr
+### Added
+* Add `Promise` to the API if callbacks are omitted (@juliangruber)
+* Add Greenkeeper badge (@ralphtheninja)
+* Add tests for `maybeError()` calling back synchronously if db is closed (@ralphtheninja)
 
-## 0.6.1 @ 1 Mar 2013
+### Changed
+* Update to `deferred-leveldown@2.0.0` (@ralphtheninja)
+* Change `levelup` constructor to take store as first parameter (@ralphtheninja)
+* Switch to use `AbstractLevelDOWN#status` (@ralphtheninja)
+* Update copyright year to 2017 (@ralphtheninja)
+* Rename `lib/util.js` to `lib/promisify.js` (@ralphtheninja)
 
-  * internal code cleanup & refactoring @ralphtheninja
-  * fix multiple iterator.end() calls in ReadStreams throwing errors (destroy() called while read/next is in progress) #82 #83 #84 @rvagg
+### Removed
+* Remove `approximateSize()` (@ralphtheninja)
+* Remove `destroy()` (@ralphtheninja)
+* Remove `repair()` (@ralphtheninja)
+* Remove `getProperty()` (@ralphtheninja)
+* Remove `.errorIfExists` (@ralphtheninja)
+* Remove `.createIfMissing` (@ralphtheninja)
+* Remove `.compression` (@ralphtheninja)
+* Remove `.cacheSize` (@ralphtheninja)
+* Remove `.sync` (@ralphtheninja)
+* Remove `.fillCache` (@ralphtheninja)
+* Remove optional `leveldown` (pass in as `options.db`) (@ralphtheninja)
```

Comment:
Did you mean "Passed in as"?

Reply:
Don't know what I mean here, basically took the commit message as is.

Reply:
I think I'll just remove the `(pass in as ..)` completely